src/tree-manager/flatten-tree.js

```diff
@@ -127,7 +136,8 @@ function setInitialStateProps(node, parent = {}) {
   }
 }
 
-function walkNodes({ nodes, list = new Map(), parent, depth = 0, simple, showPartialState, defaultValues = [], hierarchical }) {
+function walkNodes({
+  nodes, parent, depth = 0, simple, radio, showPartialState, hierarchical, returnValues = { list: new Map(), defaultNodes: [], checkedNodes: [] } }) {
```

Comment:
`returnValues` - interesting approach!

Reply:
A bit odd but thought it would make sense to group the recursive props.
Perhaps should be prefixed with _?

Reply:
Definitely makes sense. No need for underscore though. I'm wondering if this is a code smell that is telling us that perhaps we should not use recursion or maybe actually use the returned values in the next recursive call? But that's for another day. Let's go with this for now.